src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5576,34 +5576,72 @@ internal static List<CompletionResult> CompleteMember(CompletionContext context,
                     targetExpr = nextToLastAst as ExpressionAst;
                 }
             }
-            else if (lastAst.Parent is MemberExpressionAst)
+            else if (lastAst.Parent is MemberExpressionAst parentAsMemberExpression)
             {
-                // If 'targetExpr' has already been set, we should skip this step. This is for some member completion
-                // cases in ISE. In ISE, we may add a new statement in the middle of existing statements as follows:
-                //     $xml = New-Object Xml
-                //     $xml.
-                //     $xml.Save("C:\data.xml")
-                // In this example, we add $xml. between two existing statements, and the 'lastAst' in this case is
-                // a MemberExpressionAst '$xml.$xml', whose parent is still a MemberExpressionAst '$xml.$xml.Save'.
-                // But here we DO NOT want to re-assign 'targetExpr' to be '$xml.$xml'. 'targetExpr' in this case
-                // should be '$xml'.
-                if (targetExpr == null)
+                if (lastAst is ErrorExpressionAst)
                 {
-                    var memberExprAst = (MemberExpressionAst)lastAst.Parent;
-                    targetExpr = memberExprAst.Expression;
+                    // Handles scenarios like $PSVersionTable.PSVersi^.Major.
+                    // where the cursor is moved back to a previous member expression while
+                    // there's an incomplete member expression at the end
+                    targetExpr = parentAsMemberExpression;
+                    do
+                    {
+                        if (targetExpr is MemberExpressionAst memberExpression)
```

Comment:
We could move this in `while` condition.

Reply:
No we can't. It's a Do-while loop so the while condition can't set the variable.